app/src/ereferrals/java/org/eyeseetea/malariacare/domain/usecase/LoginUseCase.java

```diff
@@ -33,10 +35,21 @@ public void onError(Throwable throwable) {
                         } else if (throwable instanceof InvalidCredentialsException) {
                             callback.onInvalidCredentials();
                         } else if (throwable instanceof NetworkException) {
-                            callback.onNetworkError();
-                        } else {
-                            throwable.printStackTrace();
-                            callback.onUnexpectedError();
+                            CheckCredentialsWithOrgUnitUseCase checkCredentialsWithOrgUnitUseCase =
+                                    new CheckCredentialsWithOrgUnitUseCase();
+                            checkCredentialsWithOrgUnitUseCase.execute(
```

Comment:
A Use Case should not invoke another use case, when this happens it should be a clue that something in the design is not right